core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -2051,4 +2055,59 @@ class GroupMetadataManagerTest {
     group.transitionTo(CompletingRebalance)
     expectMetrics(groupMetadataManager, 1, 0, 1)
   }
+
+  def addDelay(durationMs: Long)(groupMetadata: GroupMetadata): Unit ={
+    time.sleep(durationMs)
+  }
+
+  @Test
+  def testPartitionLoadMetric(): Unit = {
```

Comment:
It looks like this test introduces multiple delays of several seconds, resulting in this test taking over a minute to run. This isn't good - unit tests should be very quick. We need to think of ways to cut down on the waiting and still cover the code changes.

Reply:
This is a valid issue. I removed all the `time.sleep()` calls that would cause it to run for a long time. It should not test whether metric values are calculated correctly as this should be done in `MetricsTest.java`